packages/controls/src/widget_image.ts

```diff
@@ -44,11 +44,21 @@ class ImageView extends DOMWidgetView {
          * Called when the model is changed.  The model may have been
          * changed by another view or by a state update from the back-end.
          */
-        var blob = new Blob([this.model.get('value')], {type: `image/${this.model.get('format')}`});
-        var url = URL.createObjectURL(blob);
+
+        var url;
+        var format = this.model.get('format');
+        var value = this.model.get('value');
+        if (format != 'url') {
+            var blob = new Blob([value], {type: `image/${this.model.get('format')}`});
+            url = URL.createObjectURL(blob);
+        } else {
+            url = String.fromCharCode.apply(null, new Uint8Array(value.buffer));
+        }
+
+        // Clean up the old objectURL
         var oldurl = this.el.src;
         this.el.src = url;
-        if (oldurl) {
+        if (oldurl && typeof oldurl !== 'string') {
```

Comment:
quite tricky, but createObjectURL does create a string object. Maybe sth like:
```javascript
if (oldurl && oldurl.startsWith('blob:')) {
 ...
}
```

Reply:
Hm... If it's just a string does it still need to be freed? Or is it just cast to a string when accessed through `this.el.src`?

Reply:
It is a string of the form, `'blob:<someid>'`, see https://github.com/jupyter-widgets/ipywidgets/pull/1685 how I solved it for the videostream.

Reply:
Yeah, I played around with it in the console and I see what's going on. The `src` attribute is a string that points to the location of the resource, which in this case is an in-memory blob.

I think the only downside of this `startsWith('blob:')` approach is that a user might create *another* blob that they *don't* want revoked, and pass that one's resource locator to the model, and when they change the image, we'll be freeing their blob, not *our* blob. This seems like an extreme edge case, though, and I'm sure there are simple enough workarounds that dealing with this edge case is not worth adding additional complexity.

I imagine that the "right" way to do this (which avoids this edge case) would be to set a flag on the model that indicates whether the current URL was allocated by `update`, and if so free it - rather than trying to infer this from the value at all.

Reply:
I don't think users have access to the `blob:... ` strings, so that would indeed be rather unlikely.
The [docs](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL) are a big vague, but it seems to be its reference counting behind the scenes, so I wouldn't worry about it.